exporters/secret/exporter.go

```diff
@@ -102,14 +111,69 @@ func (e *Exporter) calculateExpiry(ch chan<- prometheus.Metric, secret v1.Secret
 
 		for _, cert := range certs {
 			timestamp := float64(cert.NotAfter.Unix())
-			ch <- prometheus.MustNewConstMetric(e.cert, prometheus.GaugeValue, timestamp, secretName, secretNamespace, certKey, certType)
+			issuer := cert.Issuer.String()
+			ch <- prometheus.MustNewConstMetric(e.cert, prometheus.GaugeValue, timestamp, secretName, secretNamespace, certKey, issuer, certCRName)
 		}
 	}
 	e.logger.Log("info", fmt.Sprintf("added secret %s/%s to the metrics", secretNamespace, secretName))
 
 	return nil
 }
 
+func (e *Exporter) certificateGroupVersionResources() ([]schema.GroupVersionResource, error) {
+	groupVersions := []schema.GroupVersionResource{}
+
+	_, resources, err := e.k8sClient.ServerGroupsAndResources()
```

Comment:
Is it possible to filter here? We might get _a lot_ of responses here depending on the amount of CRDs.

Reply:
Checked the k8s.io/  code, looks like [their solution](https://github.com/kubernetes/client-go/blob/v0.21.3/discovery/helper.go#L87-L102) looks a little like mine, but a direct filter is not possible.

I could work from the [`ServerGroupsAndResources`](https://github.com/kubernetes/client-go/blob/v0.21.3/discovery/discovery_client.go#L255-L283) backwards and see if this can be slimmed down, but for now I would say this works

Reply:
That is unfortunate :/ 
We should test this and see if it has a performance impact.

Reply:
I'll prepare something and report back